include/nbl/video/IGPUQueue.h

```diff
@@ -27,6 +27,17 @@ class NBL_API IGPUQueue : public core::Interface, public core::Unmovable
             IGPUSemaphore*const * pSignalSemaphores = nullptr;
             uint32_t commandBufferCount = 0u;
             IGPUCommandBuffer*const * commandBuffers = nullptr;
+
+            inline bool isValid()
+            {
+                if(waitSemaphoreCount > 0u && (pWaitSemaphores == nullptr || pWaitDstStageMask == nullptr))
+                    return false;
+                if(signalSemaphoreCount > 0u && pSignalSemaphores == nullptr)
+                    return false;
```

Comment:
its valid to not wait or not signal any semaphores.

For example if you split some work into 5 submits in a row (due to buffering), you're guaranteed that they'll execute in submission order therefore they don't need semaphores between them.

Reply:
Yea, but this is basically saying if the `Count` is more than zero then the pointer shouldn't be null :D
If the count is zero it's ok

Reply:
don't worry, It was just a misread :D